src/main/java/gyro/google/compute/AbstractHealthCheckResource.java

```diff
@@ -0,0 +1,347 @@
+            setHttpHealthCheck(httpHealthCheck);
+        }
+
+        setHttpsHealthCheck(null);
+        if (healthCheck.getHttpsHealthCheck() != null) {
+            HealthCheckHttpsHealthCheck httpsHealthCheck = newSubresource(HealthCheckHttpsHealthCheck.class);
+            httpsHealthCheck.copyFrom(healthCheck.getHttpsHealthCheck());
+            setHttpsHealthCheck(httpsHealthCheck);
+        }
+
+        setHttp2HealthCheck(null);
+        if (healthCheck.getHttp2HealthCheck() != null) {
+            HealthCheckHttp2HealthCheck http2HealthCheck = newSubresource(HealthCheckHttp2HealthCheck.class);
+            http2HealthCheck.copyFrom(healthCheck.getHttp2HealthCheck());
+            setHttp2HealthCheck(http2HealthCheck);
+        }
+
+        setSslHealthCheck(null);
+        if (healthCheck.getSslHealthCheck() != null) {
+            HealthCheckSslHealthCheck sslHealthCheck = newSubresource(HealthCheckSslHealthCheck.class);
+            sslHealthCheck.copyFrom(healthCheck.getSslHealthCheck());
+            setSslHealthCheck(sslHealthCheck);
+        }
+
+        setTcpHealthCheck(null);
+        if (healthCheck.getTcpHealthCheck() != null) {
+            HealthCheckTcpHealthCheck tcpHealthCheck = newSubresource(HealthCheckTcpHealthCheck.class);
+            tcpHealthCheck.copyFrom(healthCheck.getTcpHealthCheck());
+            setTcpHealthCheck(tcpHealthCheck);
+        }
+    }
+
+    public HealthCheck getHealthCheck(Set<String> changedFieldNames) {
+        boolean isUpdate = changedFieldNames != null && (changedFieldNames.size() > 0);
+
+        HealthCheck healthCheck = new HealthCheck();
+
+        if (!isUpdate) {
+            healthCheck.setName(getName());
+        }
+
+        if (!isUpdate || changedFieldNames.contains("check-interval-sec")) {
+            healthCheck.setCheckIntervalSec(getCheckIntervalSec());
+        }
+
+        if (!isUpdate || changedFieldNames.contains("description")) {
+            healthCheck.setDescription(getDescription());
+        }
+
+        if (!isUpdate || changedFieldNames.contains("healthy-threshold")) {
+            healthCheck.setHealthyThreshold(getHealthyThreshold());
+        }
+
+        if (!isUpdate || changedFieldNames.contains("timeout-sec")) {
+            healthCheck.setTimeoutSec(getTimeoutSec());
+        }
+
+        if (!isUpdate || changedFieldNames.contains("unhealthy-threshold")) {
+            healthCheck.setUnhealthyThreshold(getUnhealthyThreshold());
+        }
+
+        if (isUpdate) {
+            healthCheck.setHttpHealthCheck(Data.nullOf(HTTPHealthCheck.class));
+        }
+
+        if (getHttpHealthCheck() != null) {
+            healthCheck.setType(getHttpHealthCheck().getType());
+            healthCheck.setHttpHealthCheck(getHttpHealthCheck().toHttpHealthCheck());
+        }
+
+        if (isUpdate) {
+            healthCheck.setHttpsHealthCheck(Data.nullOf(HTTPSHealthCheck.class));
+        }
```

Comment:
This check is redundant as the same operation can be performed during create as well.